lib/camera-scanner.js

```diff
@@ -44,12 +46,22 @@ export class CameraScanner extends EventEmitter {
       throw new Error('scanMode must be "first", "all", or "exhaustive"');
     }
 
+    // Validate formats if provided
+    if(formats !== null) {
+      const validFormats = ['qr_code', 'pdf417', 'pdf417_enhanced', 'mrz'];
+      const invalidFormats = formats.filter(f => !validFormats.includes(f));
```

Comment:
```suggestion
      const invalidFormats = formats.filter(format => !validFormats.includes(format));
```

Avoid using single character variable names unless they are throw-away counters. It causes people reviewing your code, and future-you, to have to expend more effort to understand what each variable refers to.

Reply:
addressed in this commit - https://github.com/digitalbazaar/bedrock-web-optical-scanner/pull/5/commits/2a8110dacdaeb265450ce78a8557a58d8d1d9297